yugabyte/src/yugabyte/ysql/client.clj

```diff
@@ -57,6 +58,17 @@
        :close close-conn
        :log?  true})))
 
+(defprotocol YSQLYbClient
+  "Used by defclient macro in conjunction with jepsen.client/Client specifying actual logic"
+  (setup-cluster! [client test c conn-wrapper]
+    "Called once on a random node to set up database/cluster state for testing.")
+  (invoke-inner! [client test operation c conn-wrapper]
```

Comment:
Is it better to rename to avoid `inner` word because it is confusing and the wrapping is not part of `YSQLYbClient` interface, but part of `defclient` macro.